### DISCUSSION TITLE
Bad file descriptor and no logs running Saleor on ECS Fargate

### DISCUSSION THREAD

**Discussion by lucaam** [saleor/saleor#8879] [Deployment]
Hello everyone,

I am currently running Saleor 3.0.0-b.36 on AWS and I configured the Core on ECS using Fargate and both Dashboard (3.0.0-b.24) and Storefront (3.0.0-a.5) on S3.

After "saleor" database creation on RDS (Postgres 12.8), I did the migration and I created the superuser so I can access the dashboard. Configured the "Site settings" and tried to do some other configuration like create wharehouse, invite new staff member (I configured SMTP with GMAIL via env variable) and so on but 99% of times I try to do some operation I get an error on the dashboard and the container stops with this error message:

`[2021-12-29 00:12:24 +0000] [55] [INFO] Error while closing socket [Errno 9] Bad file descriptor`

The main problem is that I don't see any other significant log as you can see from the image below.

![image](https://user-images.githubusercontent.com/21973580/147616948-10be1866-267d-4044-af1e-dde5a7464916.png)

Does anyone know how can I investigate on this error? Why there is no log on CloudWatch? I also tried to follow the log stream using ecs-cli but I have the exact same result.

This is currently a test environment and it is available to do some testing if anyone can.

PS:
List of what I configured:

- Core 3.0.0-b.36 (ECS Fargate - 1 vCPU, 2GB RAM)
- Dashboard 3.0.0-b.24 (S3)
- Storefront 3.0.0-a.5 (S3)
- Postgres 12.8 (t2 micro)
- Redis 6.2.5 (t2 micro) 
- Link with Sentry.io for frontend
- SMTP using GMAIL
- Static and Media on the same S3 Bucket
- Openexchangerates with Free tier
- Vatlayer with Free tier

Core:
```
DASHBOARD_URL : http://dashboard.myproject.s3-website.eu-central-1.amazonaws.com/
EMAIL_URL : smtp://info@myproject:password@smtp.gmail.com:465/?ssl=True
ALLOWED_CLIENT_HOSTS : *
CELERY_BROKER_URL : redis://test:password@redis.cache.amazonaws.com:6379/1
OPENEXCHANGERATES_API_KEY : apikey
AWS_MEDIA_CUSTOM_DOMAIN : myprojectstatic.s3.eu-central-1.amazonaws.com
AWS_SECRET_ACCESS_KEY : mysecrect
SECRET_KEY : secret
AWS_STATIC_CUSTOM_DOMAIN : myprojectstatic.s3.eu-central-1.amazonaws.com
DEBUG : True
AWS_QUERYSTRING_AUTH : True
AWS_QUERYSTRING_EXPIRE : 3600
DEFAULT_CURRENCY : EUR
DATABASE_URL : postgres://user:password@db.eu-central-1.rds.amazonaws.com:5432/saleor
AWS_ACCESS_KEY_ID : accesskey
AWS_MEDIA_BUCKET_NAME : myprojectstatic
STOREFRONT_URL : http://storeurl.eu-central-1.amazonaws.com/
ALLOWED_HOSTS : *
CREATE_IMAGES_ON_DEMAND : False
DEFAULT_FROM_EMAIL : info@myproject
DEFAULT_COUNTRY : IT
AWS_STORAGE_BUCKET_NAME : myprojectstatic
```
Storefront
```
# common envs
NEXT_PUBLIC_API_URI=http://saleor-abcd.eu-central-1.elb.amazonaws.com/graphql/
NEXT_PUBLIC_SALEOR_CHANNEL_SLUG=default-channel
NEXT_PUBLIC_DEMO_MODE=false
NEXT_PUBLIC_GTM_ID=GTM-P*****M
NEXT_PUBLIC_SENTRY_APM=0
NEXT_PUBLIC_SENTRY_DSN=
# ssr envs
INCREMENTAL_STATIC_REGENERATION_REVALIDATE=3600 # 1 hour
NEXT_PUBLIC_STATIC_PATHS_FALLBACK=blocking
```

Dashboard
```
API_URI=http://saleor-abcd.eu-central-1.elb.amazonaws.com/graphql/
API_URI to: http://localhost:8000/graphql/.
APP_MOUNT_URI=/dashboard/
STATIC_URL=http://dashboard.myproject.eu-central-1.amazonaws.com/
# Required
SENTRY_DSN=https://test@test.ingest.sentry.io/*******
# Optional
SENTRY_AUTH_TOKEN=mytoken
SENTRY_ORG=myorg
SENTRY_PROJECT=myproject
SENTRY_URL_PREFIX=
ENVIRONMENT=dev

```

**Comment by KristofTomase** [comment]
Hej Luca,
Hope You are fine :-)
I have the same enviroment  ... and i have   a lite question about createsuperuser ... how and where did you create task  for createsuperuser ?
Best regards,
KRS

**Reply by lucaam** [reply]
> Hej Luca, Hope You are fine :-) I have the same enviroment ... and i have a lite question about createsuperuser ... how and where did you create task for createsuperuser ? Best regards, KRS

Hello,

I created another task, that defined the same way of the task I use for the core but with 2 adding:
- Environment variable = "name": "DJANGO_SUPERUSER_PASSWORD" where its value is the password for the superuser
- A Command that is equals to `python,manage.py,createsuperuser,--email,myeamil@myorg.com,--noinput`

![image](https://user-images.githubusercontent.com/21973580/147824986-b3fe1cda-f8d7-4c07-ba34-b99dc732275f.png)
![image](https://user-images.githubusercontent.com/21973580/147824965-9054632b-6cd3-43eb-9f06-ac0503b59b2c.png)

Another way I used during my tests was accessing the shell after the main container was running and execute the command from there.
That's the same exact way I use to do the Django migration on DB.

I suppose you are not experiencing my problems because you still don't have the superuser (?). Let me know if you can see logs and if you face any problem. If it is possible please tell me more about your configuration!

**Reply by KristofTomase** [reply]
ok thank you for the quick reply.
I will check it and come back to your question  asap.
regards,
KRS

**Reply by lucaam** [reply]
> ok thank you for the quick reply. I will check it and come back to your question asap. regards, KRS

Did you figure it out?

**Reply by KristofTomase** [reply]
Unfortunately not yet ... struggling with storefront on S3 ...  ;-) i will back a.s.a.p   ;-)

**Reply by KristofTomase** [reply]
> > ok thank you for the quick reply. I will check it and come back to your question asap. regards, KRS
> 
> Did you figure it out?

Hej Lucaam ,
I'm back to You with this issue ...
List of what I configured:

Core 3.1 (ECS Fargate - 1 vCPU, 2GB RAM + domain + https)
Dashboard main 3.1 (S3 + CloudFront + domain + https)
Storefront 3.1  (S3 + Cloud Front + domain + https)
Postgres 13.4 (db.t3.micro)
Redis 6.2.5 (cache.t2.small)
Static and Media on the same S3 Bucket

Evertyhing works fine without problems with debug=True
I can log in , add new products ... etcetera ,
but when i change to debug=False ...  task/container accidentally restarts and i get the same error
CloudWatch log:
![log](https://user-images.githubusercontent.com/23246333/154817277-00a36b74-de8b-437d-a316-fa0c0211410f.jpg)



Any Idea ? maybe CORS setup ?
or Fargate/RDS/Redis required bigger Node Type ?

Did You fix this ?
Best Regards,
Kristoff.
